notifications/mailgun.js

```
@@ -0,0 +1,181 @@
+  rule.dayOfWeek = [new schedule.Range(WORK_WEEK_START_DAY, WORK_WEEK_END_DAY)];
+  rule.hour = DAILY_MAILING_HOUR;
+  rule.minute = DAILY_MAILING_MINUTE;
+
+  schedule.scheduleJob(rule, job);
+};
+
+export const addDailyMailing = () => {
+  dailyScheduler(() => {
+    console.log("Daily mailing");
+    getActivitiesForToday().then(activities => {
+      let mails = mailCreator(activities);
+      mails.forEach(email => {
+        mailSender(email.address, "Activity reminder", email.html, email.icons)
+          .then(res => console.log(`[${email.address}] massage: ${res.message} id: ${res.id}`))
+          .catch(err => console.error(`[${email.address}]error: ` + err))
+      })
+    }).catch(error => console.error("error: " + error))
+  });
+};
+
+export const duringDayScheduler = job => {
+  moment.tz.setDefault(DEFAULT_TIMEZONE);
+  let rule = new schedule.RecurrenceRule();
+  rule.dayOfWeek = [new schedule.Range(WORK_WEEK_START_DAY, WORK_WEEK_END_DAY)];
+  rule.hour = [new schedule.Range(ACTIVITIES_CHECK_START_HOUR, ACTIVITIES_CHECK_END_HOUR)];
+  rule.minute = [new schedule.Range(0, 59, ACTIVITIES_INTERVAL)];
+
+  schedule.scheduleJob(rule, job);
+};
+
+export const addDuringDayMailing = () => {
+  duringDayScheduler(() => {
+    console.log(`During day mailing: ${moment().utc().format("MMMM Do YYYY, h:mm:ss a")}`);
+    getNextActivities().then(activities => {
+      if (activities.length !== 0) {
+        let mails = mailCreator(activities);
+        mails.forEach(email => {
+          mailSender(email.address, "Activity reminder", email.html, email.icons)
+            .then(res => console.log(`[${email.address}] massage: ${res.message} id: ${res.id}`))
+            .catch(err => console.error(`[${email.address}] error: ` + err))
+        })
+      } else {
+        console.log("No activities")
+      }
+    }).catch(error => console.error("error: " + error))
+  })
+};
+
+
+export const runNotificationService = () => {
+  if(mailgunAPI && mailgunDomain) {
```

fail fast in production environment